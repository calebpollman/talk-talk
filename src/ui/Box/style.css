.app {
  height: 100vh;
  width: 100vw;
}

.container {
  flex: 1;
}
