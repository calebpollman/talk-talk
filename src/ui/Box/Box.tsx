import React from 'react';
import './style.css';

export interface BoxProps {
  children: React.ReactNode;
  style?: React.CSSProperties;
  variant?: 'app' | 'container';
}

function Box({ children, style, variant = 'container' }: BoxProps) {
  return (
    <div className={variant} style={style}>
      {children}
    </div>
  );
}

export default Box;
