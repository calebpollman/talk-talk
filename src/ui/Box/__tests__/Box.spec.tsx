import React from 'react';
import { render } from '@testing-library/react';

import Box, { BoxProps } from '..';

function TestChildren() {
  return null;
}

describe('Box', () => {
  it.each(['app', 'container'] as BoxProps['variant'][])('renders %s variant as expected', (variant) => {
    expect(
      render(
        <Box variant={variant}>
          <TestChildren />
        </Box>
      ).baseElement
    ).toMatchSnapshot();
  });
});
