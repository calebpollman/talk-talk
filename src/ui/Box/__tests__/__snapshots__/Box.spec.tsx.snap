// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders app variant as expected 1`] = `
<body>
  <div>
    <div />
  </div>
</body>
`;

exports[`Box renders container variant as expected 1`] = `
<body>
  <div>
    <div />
  </div>
</body>
`;
